tl

```diff
@@ -50,36 +59,33 @@ end
 package.path = script_path() .. "/?.lua;" .. package.path
 
 local tl = require("tl")
+local argparse = require("argparse")
 
-local tlconfig = get_config()
+local parser = argparse("tl", "A minimalistic typed dialect of Lua.")
 
-local cmd
-local filename
-
-local is_cmd = {
-   run = true,
-   check = true,
-   gen = true,
-}
-
-if is_cmd[arg[1]] then
-   cmd = arg[1]
-   filename = arg[2]
-else
-   cmd = "check"
-   filename = arg[1]
-end
+parser:option("-l --preload", "Execute the equivalent of require('modulename') before executing the tl script(s).")
+      :argname("<modulename>")
+      :count("*")
 
-local function printerr(s)
-   io.stderr:write(s .. "\n")
-end
+parser:command_target("command")
 
-if not filename then
-   printerr("usage:")
-   printerr("   tl run <file>")
-   printerr("   tl check <file>")
-   printerr("   tl gen <file>")
-   os.exit(1)
+local check_command = parser:command("check", "Type-check one or more tl script.")
+check_command:argument("script", "The tl script."):args("+")
+
+local gen_command = parser:command("gen", "Generate a Lua file for one or more tl script.")
+gen_command:argument("script", "The tl script."):args("+")
+
+local run_command = parser:command("run", "Run a tl script.")
+run_command:argument("script", "The tl script."):args(1)
+
+local args = parser:parse()
+
+local tlconfig = get_config()
+
+local cmd = args["command"]
+
+if args["preload"] then
+   tlconfig["preload_modules"] = args["preload"]
```

Comment:
This is _replacing_ the list in tlconfig, right? I think it should probably add to it instead.

Reply:
This behavior is based on `tsc` https://www.typescriptlang.org/docs/handbook/tsconfig-json.html#details

> Compiler options specified on the command line override those specified in the tsconfig.json file.

I don't mind changing it, though.

Reply:
Should be fixed in https://github.com/hishamhm/tl/pull/42/commits/ffff7ef8c42ae42c84df009baf741e34c314572d.

Reply:
> This behavior is based on tsc https://www.typescriptlang.org/docs/handbook/tsconfig-json.html#details

Oh, I wasn't aware. Well, let's see how people react and if there's confusion, we can change it later if we find it necessary.